### PR TITLE
Add race page

### PR DESCRIPTION
**Pull request by SaamIAm**
**Basic Race page:**
includes canvas with spammy race feature of block moving across screen. 
Feel free to roast, I was not sure why Canvas is above the banner. 
This is a WIP open to tips and tricks.

**Must Haves:**
-needs win condition (when block crosses threshold) ✅ 
-needs timer/timing feature (race time) ✅ 
-needs score feature (turns race time into score) ✅ 
-needs API connection
-needs endurance feature (button cool down) ✅

**Could Haves:**
-could use restart/reset feature ✅
-could use transition smoothing between race jumps (jerky but could be aesthetic)
-could use art and better styling

### CODE REVIEW

**Comment by Chris-Johnston** [comment]
@SaamIAm resolve merge conflicts

File: PetGame/Pages/Race.cshtml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+@{
+    Layout = "~/Pages/_Layout.cshtml";
+    ViewBag.Title = "Race";
+}
+<html lang="en">
+<h1>
+    Race Page
+</h1>
+<h2>
+    Spam button race!
```
Comment (Chris-Johnston) [inline PetGame/Pages/Race.cshtml]: Weird wording.

File: PetGame/Pages/Race.cshtml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+</h2>
+
+<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+<style>
+    canvas {
```
Comment (Chris-Johnston) [inline PetGame/Pages/Race.cshtml:18]: We should not use inline stylesheets, we already have a stylesheet.

File: PetGame/Pages/Race.cshtml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    <script>
+var myGamePiece;
+
+function startGame() {
+    myGamePiece = new component(20, 20, "red", 10, 120);
+    myGameArea.start();
+}
+
+var myGameArea = {
+    canvas : document.createElement("canvas"),
```
Comment (Chris-Johnston) [inline PetGame/Pages/Race.cshtml:33]: We should avoid creating elements, and instead have them already contained in the html body on load.

File: PetGame/Pages/Race.cshtml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+
+function moveRight() {
+    var strength = 100;
+    var distance = 600 * (strength/1000);
+    myGamePiece.speedX += (distance);
+
+}</script>
+    <div style="text-align:left;width:480px;">
```
Comment (Chris-Johnston) [inline PetGame/Pages/Race.cshtml:145]: again, we should set this style in the css, and not inline

File: PetGame/Pages/Race.cshtml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        <button onclick="moveRight();startCount()">Press Me</button>
```
Comment (Chris-Johnston) [inline PetGame/Pages/Race.cshtml]: we should give this button an id, instead of writing js in the `onclick` handler.

**Review by Chris-Johnston — CHANGES_REQUESTED** [review]
Some changes need to be made, but we do need to merge this in eventually so that it can be used.

**Comment by SaamIAm** [comment]
@Chris-Johnston How many of those comments above are mission critical?

**Comment by Chris-Johnston** [comment]
I would greatly prefer not having to go back to fix them later.

File: PetGame/Pages/Race.cshtml (hunk 5 above)
Comment (SaamIAm) [reply PetGame/Pages/Race.cshtml]: I gave it an Id but am still calling the functions onclick

**Pull request merged**